instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/jmx/BeanFinder.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.runtimemetrics.jmx;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.management.MBeanServer;
+import javax.management.MBeanServerFactory;
+import javax.management.ObjectName;
+
+/**
+ * A class responsible for finding MBeans that match metric definitions specified by a set of
+ * MetricDefs.
+ */
+class BeanFinder {
+
+  private final MetricRegistrar registrar;
+  private MetricConfiguration conf;
+  private final ScheduledExecutorService exec = Executors.newSingleThreadScheduledExecutor();
+  private final long exportInterval;
+  private final long maxDelay;
+  private long delay = 1000; // number of milliseconds until first attempt to discover MBeans
+
+  BeanFinder(MetricRegistrar registrar, ConfigProperties configProperties) {
+    this.registrar = registrar;
+    this.exportInterval = configProperties.getLong("otel.metric.export.interval", 60000);
```

Comment:
Why tie the bean discovery interval to the export interval?

Reply:
The thinking is that bean discovery can get expensive - a lot of parsing with regex matching. So it makes little sense to pay the price of bean discovery more often than the found metrics could be observed from the outside.

Reply:
Sure, but having the discovery interval tied to the export interval seems surprising. Maybe they should share the same default value, but be separately configurable.

Reply:
Good point, I'll change that